### PR TITLE
Escape underscores tex

### LINKED ISSUE #1228: [TASK] Handle underscore nodes in TeX
--------
Issue Description
--------
When the InputData specs write their LaTeX for manual purposes, if a node name has an underscore, and that node has attributes or subnodes, e.g.
```xml
<node_name type='abc'>1</node_name>
```
there can be a resulting failure to escape underscores; namely, you get something like
```
The node \xmlNode{node\\_name} ...
```
instead of
```
The node \xmlNode{node\_name} ...
```
This fails to compile in TeX.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
Closes #1228

Assures that, when escaping underscores in LaTeX, a second escaping backslash isn't erroneously added.

Note this occasion does not arise in RAVEN, but it does come up in HERON, where this change fixes the problem. I could not devise a straightforward test for this change.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/utils/InputData.py [hunk 1]
```diff
@@ -425,7 +426,8 @@ def generateLatex(cls, recDepth=0):
       msg += '{i}\\end{{itemize}}\n'.format(i=doDent(recDepth, 1))
     # TODO is this a good idea? -> disables underscores in math mode :(
     if recDepth == 0:
-      msg = msg.replace('_', '\_')
+      # assure underscore is escaped, but not doubly
+      msg = re.sub(r'(?<!\\)_', '\_', msg)
```
[the remark this answers is not in the document]
Comment (PaulTalbot-INL) [reply framework/utils/InputData.py]: Okay, using the substitution string r'\_' instead of '\_'.

File: framework/utils/InputData.py [hunk 2]
```diff
@@ -427,7 +427,7 @@ def generateLatex(cls, recDepth=0):
     # TODO is this a good idea? -> disables underscores in math mode :(
     if recDepth == 0:
       # assure underscore is escaped, but not doubly
-      msg = re.sub(r'(?<!\\)_', '\_', msg)
+      msg = re.sub(r'(?<!\\)_', 'r\_', msg)
```
Comment (joshua-cogliati-inl) [inline framework/utils/InputData.py:430]: Opps, r should be outside the quote marks.

Comment (PaulTalbot-INL) [reply framework/utils/InputData.py]: true, sorry

Comment (joshua-cogliati-inl) [reply framework/utils/InputData.py]: Fixed.

File: framework/utils/InputData.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshua-cogliati-inl) [reply framework/utils/InputData.py]: Fixed.

**Pull request merged**